### ISSUE TITLE
TypeError Cannot read property 'emitter' of undefined

### ISSUE THREAD

**Issue by PeterKnight** [pubkey/broadcast-channel#6]
I'm getting the following error and I'm not sure why?

```
/sandbox/node_modules/broadcast-channel/src/methods/node.js:617
    channelState.socketEE.emitter.removeAllListeners();
                          ^

TypeError: Cannot read property 'emitter' of undefined
    at close (/sandbox/node_modules/broadcast-channel/src/methods/node.js:617:27)
    at state.removeUnload.unload.add (/sandbox/node_modules/broadcast-channel/src/methods/node.js:386:40)
    at /sandbox/node_modules/unload/dist/lib/index.js:51:19
    at Set.forEach (<anonymous>)
    at runAll (/sandbox/node_modules/unload/dist/lib/index.js:50:13)
    at process.<anonymous> (/sandbox/node_modules/unload/dist/lib/node.js:38:12)
    at emitOne (events.js:116:13)
    at process.emit (events.js:211:7)
    at process._fatalException (bootstrap_node.js:378:26)
    at process.<anonymous> (/sandbox/node_modules/esm/esm.js:1:245197)

```
This happens when using RxDB in node to create a database while using node-websql, esm and babel-polyfill. It doesn't happen on the first time it creates the database, but on subsequent loads.

I've reproduced the error in a sandbox over here. Note that the error might not show until the sandbox is reloaded for the 2nd time.

[https://codesandbox.io/s/ko98889prv](https://codesandbox.io/s/ko98889prv)

**Comment by pubkey** [comment]
Hi @PeterKnight thanks for reporting.
I restarted the codesandbox multiple times and also downloaded the repo and tried it locally, but I could not reproduce the error.

**Comment by PeterKnight** [comment]
That's odd, thank you for looking though. When I'm loading the codesandbox test it still produces the error for me (see screenshot below). I get the same error locally as well, so I figured it's not just something about my machine. Hmm. Not sure what to do.

![2018-11-04 21_27_38-ko98889prv - codesandbox - opera](https://user-images.githubusercontent.com/906292/47970095-b7d9c600-e078-11e8-9ea4-12760e983ded.png)

**Comment by pubkey** [comment]
Which node-version do you use?

**Comment by PeterKnight** [comment]
node v9.9.0 on a Windows x64 machine

**Comment by PeterKnight** [comment]
I've adjusted the test sandbox code so it produces the error instantaneously by creating a database of the same name twice, could you try this locally to see if you get the error now? Code can be downloaded from the same sandbox url I used before
[https://codesandbox.io/s/ko98889prv](https://codesandbox.io/s/ko98889prv)

**Comment by pubkey** [comment]
Opening the codesandbox gives me "Error: Reloading too fast."
Also I redownloaded it and install node 9.9.0 but could still not reproduce the error.
Can you create a test that reproduces this problem and also crashes in the CI? You could start by editing [this file](https://github.com/pubkey/rxdb/blob/master/test/unit/bug-report.test.js)

**Comment by PeterKnight** [comment]
Sorry about that, codesandbox server sandboxes still a bit buggy I think.

I've tried running the unit tests but I'm getting unrelated errors trying to get that to go. 

I have been able to recreate the error on repl.it here, it should show up after it has finished installing packages and hitting run.

[https://repl.it/@PeterKnight/SecretWellgroomedLine?language=nodejs](https://repl.it/@PeterKnight/SecretWellgroomedLine?language=nodejs)

The code is really simple, and only using esm, pouchdb-adapter-node-websql, rxjs and rxdb packages.

`import * as RxDB from "rxdb";
import * as pouchAdapterNodeWebsql from "pouchdb-adapter-node-websql";

RxDB.plugin(pouchAdapterNodeWebsql);

process.stdout.write('creating test database');

const testDB = RxDB.create({
  name: "test",
  adapter: "websql",
  multiInstance: true,
  ignoreDuplicate: true
});

testDB.then(function(db) {
  process.stdout.write('database loading finished' );
});
`
